### DISCUSSION TITLE
how to extract image from stage.children

### DISCUSSION THREAD

**Discussion by claudewowo** [pixijs/pixijs#8348] [Q&A]
### Expected Behavior
extract image in stage.children, with width of this image, not the stage's.
### Current Behavior
app.renderer.plugins.extract.base64(stage.children[1]), i got a string, but it's width is the stage's, I have two questions:
1. the origin image
2. if it's cropped, how to get the new image?

### Environment
**`pixi.js` version**: 6.3.2
**Browser & Version**: Chrome 101
**OS & Version**: MacOS 12.3

**Comment by bigtimebuddy** [comment]
I'd suggest using a RenderTexture to layout exactly how you'd like, then export the render texture instead.

**Reply by claudewowo** [reply]
can you show a demo for me? thank u.

**Reply by claudewowo** [reply]
I had a demo, could you help me? 

https://www.pixiplayground.com/#/edit/QR8otayYYujJWGH-mmHD8

**Reply by claudewowo** [reply]
Hi, I create a new canvas, drawImage the sprite.texture.baseTexture.resource.source to it, after transformed the image, I create a new texture to old sprite, and it's worked! like this:

`js
      const obj = app.stage.children[0];
      const { originWidth, originHeight, originAngle } = obj;
      const image = obj.texture.baseTexture.resource.source;
      const canvas = document.createElement('canvas');

      canvas.width = originWidth;
      canvas.height = originHeight;
      canvas.style.width = originWidth;
      canvas.style.height = originHeight;
      canvas.style.transform = `rotate(${originAngle}deg)`;
      canvas.style.transformOrigin = `${originWidth / 2}px ${originHeight / 2}px`;

      document.body.appendChild(canvas);

      const ctx = canvas.getContext('2d');

      ctx.translate(originWidth / 2, originHeight / 2);
      ctx.rotate(30 / 180 * Math.PI);
      ctx.scale(obj.width / originWidth, obj.height / originHeight);
      ctx.drawImage(image, -originWidth/2, -originHeight/2, originWidth, originHeight);

      const url = canvas.toDataURL('image/png');
      const texture = new this.PIXI.Texture.from(url);

      obj.texture = texture;
      obj.angle = originAngle;
      obj.width = originWidth;
      obj.height = originHeight;
`

But it seems stubid, I want to know a better way, can you give me some suggestion?
